pkg/kudoctl/util/kudo/kudo.go

```diff
@@ -42,44 +45,40 @@ func NewClient(kubeConfigPath string, requestTimeout int64) (*Client, error) {
 	// set default configs
 	config.Timeout = time.Duration(requestTimeout) * time.Second
 
-	// create the clientset
-	kudoClientset, err := versioned.NewForConfig(config)
+	kubeClient, err := kube.GetKubeClient(kubeConfigPath)
 	if err != nil {
-		return nil, err
-	}
-
-	// use the apiextensions clientset to check for the existence of KUDO CRDs in the cluster
-	extensionsClientset, err := extensionsclient.NewForConfig(config)
-	if err != nil {
-		return nil, err
+		return nil, clog.Errorf("could not get Kubernetes client: %s", err)
 	}
 
-	_, err = extensionsClientset.CustomResourceDefinitions().Get("operators.kudo.dev", v1.GetOptions{})
+	err = kudoinit.ValidateManager(kubeClient, kudoinit.NewOptions("", ""))
 	if err != nil {
-		// timeout is not a wrappable error, timeout is an underlying issue that is NOT CRD specific, there is no value in wrapping or converting as well.
-		// best to provide the actual error for proper reporting.
+		// timeout is not a wrappable error, timeout is an underlying issue that is NOT validation specific,
+		// there is no value in wrapping or converting as well. best to provide the actual error for proper reporting.
 		if os.IsTimeout(err) {
 			return nil, err
 		}
-		return nil, fmt.Errorf("operators crd: %w", err)
+		clog.V(0).Printf("KUDO manager not correctly installed. Do you need to run kudo init?")
+		if validateInstall {
```

Comment:
Hmm, so at this point the validation has already been done and we error out if validation was enabled. We should rather only run the validation if it is requested, because the validation might be an expensive operation.

That said, validation isn't expensive right now and we shouldn't optimize prematurely, we can keep this as-is, your call :)

Reply:
Well, my intention with the validate-install parameter was more in the realm of "I know, my KUDO ctrl does not match the installed version, I still know what i'm doing and want to execute this" instead of saving the overhead of validation.

It's good thing to know that you're not matching the version, but you still can do stuff (at your own risk)

Reply:
Oh, I missed the part where `validate-install` is true by default. That makes sense, yes.